File: 03_recursion/ruby/02_greet.rb
def greet2(name)
  puts "how are you, #{name}?"
end

def bye
  print "ok bye!"
end

def greet(name)
  puts "hello, #{name}!"
  greet2(name)
  puts "getting ready to say bye..."
  bye
end

greet("adit")

File: 03_recursion/elixir/02_greet.exs
defmodule Greeting do
  def greet2(name) do
    IO.puts("how are you, #{name}?")
  end

  def bye do
    IO.puts("ok bye!")
  end

  def greet(name) do
    IO.puts("hello, #{name}!")
    greet2(name)
    IO.puts("getting ready to say bye...")
    bye()
  end
end

Greeting.greet("adit")
